frontend/src/components/CompareAssignmentsTab/CompareAssignmentsTab.js

```diff
@@ -1,69 +1,141 @@
+      return data.heatMapData;
+    case 'By Sections':
+      return data.heatMapDataWithSections;
+    default:
+      return data.heatMapDataWithSections;
+  }
+};
 
 const CompareAssignmentsTab = () => {
   const { state, dispatch } = useContext(AppContext);
 
   // TODO: these transformations and updates to state need to be re-evaluated
-  if (state.processing.loadingBusinessData === false
+  if (
+    state.processing.loadingBusinessData === false
     && state.processing.pivotedHeatMap === false
-    && state.processing.pivotingHeatMap === false) {
-    // TODO : make this async
+    && state.processing.pivotingHeatMap === false
+  ) {
     dispatch({ type: heatMapDataPivoting });
     const vizData = pivotHeatMapData(state.businessData);
+    const vizDataWithSections = pivotHeatMapDataWithSections(
+      state.businessData,
+      state.businessData.sections.map((s) => s.sis_section_id),
+    );
     dispatch({ type: heatMapDataPivoted, value: vizData });
+    dispatch({
+      type: heatMapDataWithSectionsPivoted,
+      value: vizDataWithSections,
+    });
   }
 
-  if (state.visualizationData.heatMapData.length > 0
-    && state.controls.selectors.showingRubrics.values.length === 1) {
+  if (
+    state.visualizationData.heatMapData.length > 0
+    && state.controls.selectors.showingRubrics.values.length === 1
+  ) {
     dispatch({
       type: selectorValuesUpdated,
       selectorKey: 'showingRubrics',
-      value: ['All assignments', ...state.visualizationData.heatMapData.map((r) => r.name)],
+      value: [
+        'All assignments',
+        ...state.visualizationData.heatMapData.map((r) => r.name),
+      ],
+    });
+  }
+
+  if (
+    state.visualizationData.heatMapData.length > 0
+    && state.controls.selectors.sections.values.length === 1
+  ) {
+    dispatch({
+      type: selectorValuesUpdated,
+      selectorKey: 'sections',
+      value: [
+        'All sections',
+        ...state.businessData.sections.map((s) => s.sis_section_id),
+      ],
     });
   }
 
   const filterAssignment = state.controls.selectors.showingRubrics.selected;
-  const assignmentSet = state.controls.selectors.showingRubrics.selected === 'All assignments'
-    ? (
-      state.visualizationData.heatMapData
-    ) : (
-      state.visualizationData.heatMapData.filter((rubric) => rubric.name === filterAssignment)
-    );
+  const filterSection = state.controls.selectors.sections.selected;
+  const assignmentSet = switchVizData(
+    state.controls.selectors.showSections.selected,
+    state.visualizationData,
+  );
+
+  const filteredAssignmentSet = state.controls.selectors.showingRubrics.selected === 'All assignments'
+    ? assignmentSet
+    : assignmentSet.filter((rubric) => rubric.name === filterAssignment);
+
+  let filteredSectionSet = filteredAssignmentSet;
+  if (state.controls.selectors.showSections.selected === 'By Sections') {
+    filteredSectionSet = state.controls.selectors.sections.selected === 'All sections'
```

Comment:
I like the way you progressively filter the data based on the selected assignment and section (if any). Very clear!